pkg/local/sync.go

```diff
@@ -284,8 +285,13 @@ func (s *SyncManager) upload(ctx context.Context, rootPath string, remote *uri.U
 		}
 	}()
 
+	mtimeKey := ClientMtimeHeaderKey
+	if s.presign {
+		mtimeKey = ClientMtimeMetadataKey
+	}
+
 	metadata := map[string]string{
-		ClientMtimeMetadataKey: strconv.FormatInt(fileStat.ModTime().Unix(), 10),
+		mtimeKey: strconv.FormatInt(fileStat.ModTime().Unix(), 10),
```

Comment:
can we pass the same metadata - the transport (upload presign) will detect the key type and add the header prefix to the key to transfer the headers

Reply:
Done - hope it's ok now